pyrealm/splash/splash.py

```diff
@@ -275,7 +279,7 @@ def calculate_soil_moisture(
         wn_out = np.full_like(self.tc, np.nan)
         ro_out = np.full_like(self.tc, np.nan)
 
-        curr_wn = wn_init
+        curr_wn = bounds_checker(wn_init, 0, self.kWm)
         for day_idx in np.arange(self.pn.shape[0]):
```

Comment:
This should be a value error like L238 - L240 above. The initial definition of `self.kWm` should be flexible and allow values with a warning if they seem very high - as it now is with the `bounds_checker` - but that sets hard limits and it should be an error to exceed those initial values.

Reply:
Ah sorry for missing this one and thanks for catching it! I also have updated the type annotation of `Bounds_checker`.